src/Main.java

```diff
@@ -1,16 +1,31 @@
-//TIP To <b>Run</b> code, press <shortcut actionId="Run"/> or
-// click the <icon src="AllIcons.Actions.Execute"/> icon in the gutter.
+import java.util.ArrayList;
+import java.util.List;
+import shapes.Calculable;
+import shapes.Circle;
+import shapes.ShapeFactory;
+import shapes.Square;
+import shapes.Triangle;
+
 public class Main {
 
   public static void main(String[] args) {
-    //TIP Press <shortcut actionId="ShowIntentionActions"/> with your caret at the highlighted text
-    // to see how IntelliJ IDEA suggests fixing it.
-    System.out.printf("Hello and welcome!");
 
-    for (int i = 1; i <= 5; i++) {
-      //TIP Press <shortcut actionId="Debug"/> to start debugging your code. We have set one <icon src="AllIcons.Debugger.Db_set_breakpoint"/> breakpoint
-      // for you, but you can always add more by pressing <shortcut actionId="ToggleLineBreakpoint"/>.
-      System.out.println("i = " + i);
-    }
+    System.out.println("==================================");
+    List<Calculable> list = new ArrayList<>();
+    list.add(ShapeFactory.create(Circle.class, 10, 20));
```

Comment:
is it really worth it to create a factory to create an object of a class? It is overengineering. No reason to use this pattern here